### PR TITLE
Prevent building invalid blocks

### PR DESCRIPTION
**Pull request by arkpar**
* `apply_extrinsic` return value is now considered by the block builder.
* Removed `panic=abort` since native runtime requires unwinding.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -89,6 +89,3 @@ travis-ci = { repository = "paritytech/polkadot", branch = "master" }
 maintenance = { status = "actively-developed" }
 is-it-maintained-issue-resolution = { repository = "paritytech/polkadot" }
 is-it-maintained-open-issues = { repository = "paritytech/polkadot" }
-
-[profile.release]
-panic = "abort"
```
Comment (pepyakin) [inline Cargo.toml]: Even though `panic=unwind` is the default, I'd rather specify it explicitly with a comment for documentation purposes.

File: substrate/client/src/error.rs [hunk 2]
```diff
@@ -111,6 +118,12 @@ error_chain! {
 			description("remote fetch failed"),
 			display("Remote data fetch has been failed"),
 		}
+
+		/// Error decoding extrinsic outcome.
+		ApplyExtinsicFailed {
```
Comment (pepyakin) [inline substrate/client/src/error.rs]: 1. I'm not sure why this is called `ApplyExtinsicFailed`? I think it is more like `ApplyResultDecodeFailed`
2. Why the comment, description and display messages refers to an `outcome` and not to a `result`?

File: substrate/client/src/error.rs [hunk 3]
```diff
@@ -34,6 +35,12 @@ error_chain! {
 			display("UnknownBlock: {}", &*h),
 		}
 
+		/// Extrinsic error.
+		Runtime(e: ApplyError) {
+			description("Extrinsic error"),
+			display("Extrinsic error: {:?}", e),
+		}
```
Comment (pepyakin) [inline substrate/client/src/error.rs]: Should we rename this to something more extrinsic-related. We could use `ApplyExtinsicFailed` for this IMO.

**Comment by gavofyork** [comment]
Would be nice to know why the transaction queue/block builder was letting blatantly invalid transactions through at all...

**Comment by arkpar** [comment]
I'm not sure if "insufficient balance" is blatantly invalid. And why `apply_extrinsic` would not allow them in.

**Comment by arkpar** [comment]
Transaction pool does not check for insufficient balance currently. And for signature correctness as far as I can see. 
@tomusdrw Is this right?

**Pull request merged**